pandas/core/frame.py

```diff
@@ -8065,6 +8072,15 @@ def _join_compat(
             other = DataFrame({other.name: other})
 
         if isinstance(other, DataFrame):
+            if how == "cross":
+                return merge(
+                    self,
+                    other,
```

Comment:
why do you need to do this?

Reply:
If we want to suport cross for join, we have to parametrize merge( differntly, because the validation would raise otherwise.